### ISSUE TITLE
i is not equal to j

### ISSUE THREAD

**Issue by thinkersong** [VictoriaMetrics/metricsql#24]
https://github.com/VictoriaMetrics/metricsql/blob/a271ee257023dd183d444ec7e1b363c64a3db0e3/lexer.go#L641

```
	if s[i] == '.' {
		j := i
		i++
		for i < len(s) && isDecimalChar(s[i]) {
			i++
		}
		if i == j || i == len(s) {
			return -1
		}
	}
```

I'm learning MetricsQL and I've noticed that here 'i' might not be equal to 'j' because 'i' undergoes an addition operation. Should it be j+1 here?

**Comment by Vandit1604** [comment]
You are right that `i` can never equal `j` here. `j` is set to the index of the dot and `i` is incremented on the next line, so `i` is at least `j+1` by the time the condition runs. Removing the condition leaves `j` unused, and the compiler confirms it.

On whether it should be `j+1`: I do not think so. That would reject a duration with a trailing dot, such as `-3.H`, which is valid today and covered by `TestDurationSuccess`. The same applies to plain numbers, where `234.` is accepted in `TestScanPositiveNumberSuccess`. So a duration like `1.s` meaning `1s` looks intentional rather than a parsing mistake.

That leaves the condition guarding nothing, so I opened #77 to drop it. `i == len(s)` still rejects a duration that ends with a dot and has no unit. No behaviour change, all existing tests pass.
